src/app/_components/post-card/post-card.ts

```diff
@@ -1,25 +1,75 @@
-import { Input, Component, ElementRef } from '@angular/core';
-import { Router } from '@angular/router';
-import { ActivatedRoute } from '@angular/router';
+import { Input, Component, ViewChild } from '@angular/core';
 import { ContentService } from '_services/index';
+import { ApiService } from '_services/api.service';
+import { FormBuilder, FormGroup, Validators } from '@angular/forms';
+
 
 @Component({
   selector: 'post-card',
   templateUrl: 'post-card.html',
-  styleUrls: ['post-card.scss'],
+  styleUrls: ['post-card.scss']
 })
 export class PostCardComponent {
-
   @Input('post')
   public post: any;
+
+  @ViewChild('ngFormDirective') formDirective;
+  public commentForm: FormGroup;
+
+  public comments: any;
   public isTarget: boolean;
 
   constructor (
-    private route: ActivatedRoute,
-    private elementRef: ElementRef,
     public content: ContentService,
+    public formBuilder: FormBuilder,
+    public apiService: ApiService,
   ) {
-    this.isTarget = false;
+    this.comments = [];
+  }
+
+  ngOnInit() {
+    this.getComments();
+
+    this.commentForm = this.formBuilder.group({
+      body: [''],
+    });
+  }
+
+  getComments() {
+    this.apiService.getPostComments(this.post.id).subscribe(res => {
+      this.comments = res;
+      console.log(this.comments);
+    })
+  }
+
+  commentMapLocalToApi(formValues) {
+    let comment: any = {};
+    comment.body = formValues.body;
+    return comment;
+  }
+
+  comment() {
+    let formValues = this.commentForm.value;
+    let comment = this.commentMapLocalToApi(formValues);
```

Comment:
commentMapLocalToApi isn't needed in this case. The code it was copied/modified from had significant differences in the form data structure to the api data structure. In this case it's exactly the same, so formValues could be submitted to createComment as long as the form is valid.